numpy/random/_pickle.py

```diff
@@ -1,3 +1,4 @@
+from .bit_generator import BitGenerator
```

Comment:
```suggestion
from __future__ import annotations

from .bit_generator import BitGenerator
```

Gona need a `from __future__ import annotations` statement in order to use the `|`-based union syntax prior to python 3.11. 

This is only relevant for .py files btw; for .pyi it would be redundant as nothing in there is executed during runtime.